
from PyQt4 import QtGui, QtCore

class Ui_genund(object):
    def setupUi(self, genund):
        genund.setObjectName("genund")
        genund.resize(300, 384)
        self.formLayoutWidget = QtGui.QWidget(genund)
        self.formLayoutWidget.setGeometry(QtCore.QRect(30, 30, 241, 301))
        self.formLayoutWidget.setObjectName("formLayoutWidget")
        self.formLayout = QtGui.QFormLayout(self.formLayoutWidget)
        self.formLayout.setFieldGrowthPolicy(QtGui.QFormLayout.AllNonFixedFieldsGrow)
        self.formLayout.setContentsMargins(0, 0, 0, 0)
        self.formLayout.setObjectName("formLayout")
        self.label = QtGui.QLabel(self.formLayoutWidget)
        self.label.setObjectName("label")
        self.formLayout.setWidget(0, QtGui.QFormLayout.LabelRole, self.label)
        self.iwityp = QtGui.QPushButton(self.formLayoutWidget)
        self.iwityp.setObjectName("iwityp")
        self.formLayout.setWidget(0, QtGui.QFormLayout.FieldRole, self.iwityp)
        self.label_2 = QtGui.QLabel(self.formLayoutWidget)
        self.label_2.setObjectName("label_2")
        self.formLayout.setWidget(1, QtGui.QFormLayout.LabelRole, self.label_2)
        self.aw0 = QtGui.QLineEdit(self.formLayoutWidget)
        self.aw0.setObjectName("aw0")
        self.formLayout.setWidget(1, QtGui.QFormLayout.FieldRole, self.aw0)
        self.label_3 = QtGui.QLabel(self.formLayoutWidget)
        self.label_3.setObjectName("label_3")
        self.formLayout.setWidget(2, QtGui.QFormLayout.LabelRole, self.label_3)
        self.xlamd = QtGui.QLineEdit(self.formLayoutWidget)
        self.xlamd.setObjectName("xlamd")
        self.formLayout.setWidget(2, QtGui.QFormLayout.FieldRole, self.xlamd)
        self.label_4 = QtGui.QLabel(self.formLayoutWidget)
        self.label_4.setObjectName("label_4")
        self.formLayout.setWidget(3, QtGui.QFormLayout.LabelRole, self.label_4)
        self.nwig = QtGui.QLineEdit(self.formLayoutWidget)
        self.nwig.setObjectName("nwig")
        self.formLayout.setWidget(3, QtGui.QFormLayout.FieldRole, self.nwig)
        self.label_5 = QtGui.QLabel(self.formLayoutWidget)
        self.label_5.setObjectName("label_5")
        self.formLayout.setWidget(5, QtGui.QFormLayout.LabelRole, self.label_5)
        self.label_6 = QtGui.QLabel(self.formLayoutWidget)
        self.label_6.setObjectName("label_6")
        self.formLayout.setWidget(6, QtGui.QFormLayout.LabelRole, self.label_6)
        self.label_7 = QtGui.QLabel(self.formLayoutWidget)
        self.label_7.setObjectName("label_7")
        self.formLayout.setWidget(7, QtGui.QFormLayout.LabelRole, self.label_7)
        self.delaw = QtGui.QLineEdit(self.formLayoutWidget)
        self.delaw.setObjectName("delaw")
        self.formLayout.setWidget(5, QtGui.QFormLayout.FieldRole, self.delaw)
        self.awx = QtGui.QLineEdit(self.formLayoutWidget)
        self.awx.setObjectName("awx")
        self.formLayout.setWidget(6, QtGui.QFormLayout.FieldRole, self.awx)
        self.awy = QtGui.QLineEdit(self.formLayoutWidget)
        self.awy.setObjectName("awy")
        self.formLayout.setWidget(7, QtGui.QFormLayout.FieldRole, self.awy)
        self.label_8 = QtGui.QLabel(self.formLayoutWidget)
        self.label_8.setObjectName("label_8")
        self.formLayout.setWidget(4, QtGui.QFormLayout.LabelRole, self.label_8)
        self.nsec = QtGui.QLineEdit(self.formLayoutWidget)
        self.nsec.setObjectName("nsec")
        self.formLayout.setWidget(4, QtGui.QFormLayout.FieldRole, self.nsec)

        self.retranslateUi(genund)
        QtCore.QMetaObject.connectSlotsByName(genund)

    def retranslateUi(self, genund):
        genund.setWindowTitle(QtGui.QApplication.translate("genund", "Form", None, QtGui.QApplication.UnicodeUTF8))
        self.label.setText(QtGui.QApplication.translate("genund", "Undulator Type:", None, QtGui.QApplication.UnicodeUTF8))
        self.iwityp.setText(QtGui.QApplication.translate("genund", "Planar/Helical", None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("genund", "Undulator Parameter:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_3.setText(QtGui.QApplication.translate("genund", "Period Length:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_4.setText(QtGui.QApplication.translate("genund", "Number of Periods:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_5.setText(QtGui.QApplication.translate("genund", "Field Errors:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_6.setText(QtGui.QApplication.translate("genund", "Offset - x:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_7.setText(QtGui.QApplication.translate("genund", "Offset - y:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_8.setText(QtGui.QApplication.translate("genund", "Sections:", None, QtGui.QApplication.UnicodeUTF8))

class Ui_genfodo(object):
    def setupUi(self, genfodo):
        genfodo.setObjectName("genfodo")
        genfodo.resize(289, 409)
        self.formLayoutWidget = QtGui.QWidget(genfodo)
        self.formLayoutWidget.setGeometry(QtCore.QRect(30, 30, 231, 341))
        self.formLayoutWidget.setObjectName("formLayoutWidget")
        self.formLayout = QtGui.QFormLayout(self.formLayoutWidget)
        self.formLayout.setContentsMargins(0, 0, 0, 0)
        self.formLayout.setObjectName("formLayout")
        self.label = QtGui.QLabel(self.formLayoutWidget)
        self.label.setObjectName("label")
        self.formLayout.setWidget(0, QtGui.QFormLayout.LabelRole, self.label)
        self.lineEdit = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit.setObjectName("lineEdit")
        self.formLayout.setWidget(0, QtGui.QFormLayout.FieldRole, self.lineEdit)
        self.label_2 = QtGui.QLabel(self.formLayoutWidget)
        self.label_2.setObjectName("label_2")
        self.formLayout.setWidget(1, QtGui.QFormLayout.LabelRole, self.label_2)
        self.lineEdit_2 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_2.setObjectName("lineEdit_2")
        self.formLayout.setWidget(1, QtGui.QFormLayout.FieldRole, self.lineEdit_2)
        self.label_3 = QtGui.QLabel(self.formLayoutWidget)
        self.label_3.setObjectName("label_3")
        self.formLayout.setWidget(2, QtGui.QFormLayout.LabelRole, self.label_3)
        self.lineEdit_3 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_3.setObjectName("lineEdit_3")
        self.formLayout.setWidget(2, QtGui.QFormLayout.FieldRole, self.lineEdit_3)
        self.label_4 = QtGui.QLabel(self.formLayoutWidget)
        self.label_4.setObjectName("label_4")
        self.formLayout.setWidget(3, QtGui.QFormLayout.LabelRole, self.label_4)
        self.lineEdit_4 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_4.setObjectName("lineEdit_4")
        self.formLayout.setWidget(3, QtGui.QFormLayout.FieldRole, self.lineEdit_4)
        self.label_5 = QtGui.QLabel(self.formLayoutWidget)
        self.label_5.setObjectName("label_5")
        self.formLayout.setWidget(4, QtGui.QFormLayout.LabelRole, self.label_5)
        self.lineEdit_5 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_5.setObjectName("lineEdit_5")
        self.formLayout.setWidget(4, QtGui.QFormLayout.FieldRole, self.lineEdit_5)
        self.label_6 = QtGui.QLabel(self.formLayoutWidget)
        self.label_6.setObjectName("label_6")
        self.formLayout.setWidget(5, QtGui.QFormLayout.LabelRole, self.label_6)
        self.lineEdit_6 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_6.setObjectName("lineEdit_6")
        self.formLayout.setWidget(5, QtGui.QFormLayout.FieldRole, self.lineEdit_6)
        self.label_7 = QtGui.QLabel(self.formLayoutWidget)
        self.label_7.setObjectName("label_7")
        self.formLayout.setWidget(6, QtGui.QFormLayout.LabelRole, self.label_7)
        self.label_8 = QtGui.QLabel(self.formLayoutWidget)
        self.label_8.setObjectName("label_8")
        self.formLayout.setWidget(7, QtGui.QFormLayout.LabelRole, self.label_8)
        self.lineEdit_7 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_7.setObjectName("lineEdit_7")
        self.formLayout.setWidget(6, QtGui.QFormLayout.FieldRole, self.lineEdit_7)
        self.lineEdit_8 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_8.setObjectName("lineEdit_8")
        self.formLayout.setWidget(7, QtGui.QFormLayout.FieldRole, self.lineEdit_8)
        self.label_9 = QtGui.QLabel(self.formLayoutWidget)
        self.label_9.setObjectName("label_9")
        self.formLayout.setWidget(8, QtGui.QFormLayout.LabelRole, self.label_9)
        self.label_10 = QtGui.QLabel(self.formLayoutWidget)
        self.label_10.setObjectName("label_10")
        self.formLayout.setWidget(9, QtGui.QFormLayout.LabelRole, self.label_10)
        self.lineEdit_9 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_9.setObjectName("lineEdit_9")
        self.formLayout.setWidget(8, QtGui.QFormLayout.FieldRole, self.lineEdit_9)
        self.lineEdit_10 = QtGui.QLineEdit(self.formLayoutWidget)
        self.lineEdit_10.setObjectName("lineEdit_10")
        self.formLayout.setWidget(9, QtGui.QFormLayout.FieldRole, self.lineEdit_10)

        self.retranslateUi(genfodo)
        QtCore.QMetaObject.connectSlotsByName(genfodo)

    def retranslateUi(self, genfodo):
        genfodo.setWindowTitle(QtGui.QApplication.translate("genfodo", "Form", None, QtGui.QApplication.UnicodeUTF8))
        self.label.setText(QtGui.QApplication.translate("genfodo", "Focusing Quad strength:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("genfodo", "Defocusing Quad strength:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_3.setText(QtGui.QApplication.translate("genfodo", "Focusing Quad length:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_4.setText(QtGui.QApplication.translate("genfodo", "Defocusing Quad length:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_5.setText(QtGui.QApplication.translate("genfodo", "Drift between Quads:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_6.setText(QtGui.QApplication.translate("genfodo", "Starting point:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_7.setText(QtGui.QApplication.translate("genfodo", "Offset - x:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_8.setText(QtGui.QApplication.translate("genfodo", "Offset - y:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_9.setText(QtGui.QApplication.translate("genfodo", "Solenoid field component:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_10.setText(QtGui.QApplication.translate("genfodo", "Solenoid component length:", None, QtGui.QApplication.UnicodeUTF8))

class Ui_genbeam(object):
    def setupUi(self, genbeam):
        genbeam.setObjectName("genbeam")
        genbeam.resize(323, 343)
        self.formLayoutWidget = QtGui.QWidget(genbeam)
        self.formLayoutWidget.setGeometry(QtCore.QRect(30, 30, 261, 281))
        self.formLayoutWidget.setObjectName("formLayoutWidget")
        self.formLayout = QtGui.QFormLayout(self.formLayoutWidget)
        self.formLayout.setContentsMargins(0, 0, 0, 0)
        self.formLayout.setObjectName("formLayout")
        self.label = QtGui.QLabel(self.formLayoutWidget)
        self.label.setObjectName("label")
        self.formLayout.setWidget(0, QtGui.QFormLayout.LabelRole, self.label)
        self.gamma0 = QtGui.QLineEdit(self.formLayoutWidget)
        self.gamma0.setObjectName("gamma0")
        self.formLayout.setWidget(0, QtGui.QFormLayout.FieldRole, self.gamma0)
        self.label_2 = QtGui.QLabel(self.formLayoutWidget)
        self.label_2.setObjectName("label_2")
        self.formLayout.setWidget(1, QtGui.QFormLayout.LabelRole, self.label_2)
        self.delgam = QtGui.QLineEdit(self.formLayoutWidget)
        self.delgam.setObjectName("delgam")
        self.formLayout.setWidget(1, QtGui.QFormLayout.FieldRole, self.delgam)
        self.label_3 = QtGui.QLabel(self.formLayoutWidget)
        self.label_3.setObjectName("label_3")
        self.formLayout.setWidget(2, QtGui.QFormLayout.LabelRole, self.label_3)
        self.label_4 = QtGui.QLabel(self.formLayoutWidget)
        self.label_4.setObjectName("label_4")
        self.formLayout.setWidget(3, QtGui.QFormLayout.LabelRole, self.label_4)
        self.label_5 = QtGui.QLabel(self.formLayoutWidget)
        self.label_5.setObjectName("label_5")
        self.formLayout.setWidget(4, QtGui.QFormLayout.LabelRole, self.label_5)
        self.label_6 = QtGui.QLabel(self.formLayoutWidget)
        self.label_6.setObjectName("label_6")
        self.formLayout.setWidget(5, QtGui.QFormLayout.LabelRole, self.label_6)
        self.label_7 = QtGui.QLabel(self.formLayoutWidget)
        self.label_7.setObjectName("label_7")
        self.formLayout.setWidget(6, QtGui.QFormLayout.LabelRole, self.label_7)
        self.label_8 = QtGui.QLabel(self.formLayoutWidget)
        self.label_8.setObjectName("label_8")
        self.formLayout.setWidget(7, QtGui.QFormLayout.LabelRole, self.label_8)
        self.alphax = QtGui.QLineEdit(self.formLayoutWidget)
        self.alphax.setObjectName("alphax")
        self.formLayout.setWidget(2, QtGui.QFormLayout.FieldRole, self.alphax)
        self.alphay = QtGui.QLineEdit(self.formLayoutWidget)
        self.alphay.setObjectName("alphay")
        self.formLayout.setWidget(3, QtGui.QFormLayout.FieldRole, self.alphay)
        self.emitx = QtGui.QLineEdit(self.formLayoutWidget)
        self.emitx.setObjectName("emitx")
        self.formLayout.setWidget(4, QtGui.QFormLayout.FieldRole, self.emitx)
        self.emity = QtGui.QLineEdit(self.formLayoutWidget)
        self.emity.setObjectName("emity")
        self.formLayout.setWidget(5, QtGui.QFormLayout.FieldRole, self.emity)
        self.xbeam = QtGui.QLineEdit(self.formLayoutWidget)
        self.xbeam.setObjectName("xbeam")
        self.formLayout.setWidget(6, QtGui.QFormLayout.FieldRole, self.xbeam)
        self.ybeam = QtGui.QLineEdit(self.formLayoutWidget)
        self.ybeam.setObjectName("ybeam")
        self.formLayout.setWidget(7, QtGui.QFormLayout.FieldRole, self.ybeam)
        self.label_9 = QtGui.QLabel(self.formLayoutWidget)
        self.label_9.setObjectName("label_9")
        self.formLayout.setWidget(8, QtGui.QFormLayout.LabelRole, self.label_9)
        self.curpeak = QtGui.QLineEdit(self.formLayoutWidget)
        self.curpeak.setObjectName("curpeak")
        self.formLayout.setWidget(8, QtGui.QFormLayout.FieldRole, self.curpeak)

        self.retranslateUi(genbeam)
        QtCore.QMetaObject.connectSlotsByName(genbeam)

    def retranslateUi(self, genbeam):
        genbeam.setWindowTitle(QtGui.QApplication.translate("genbeam", "Form", None, QtGui.QApplication.UnicodeUTF8))
        self.label.setText(QtGui.QApplication.translate("genbeam", "Relativistic Factor (beam energy):", None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("genbeam", "Energy spread:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_3.setText(QtGui.QApplication.translate("genbeam", "Alpha - x:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_4.setText(QtGui.QApplication.translate("genbeam", "Alpha - y:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_5.setText(QtGui.QApplication.translate("genbeam", "Emittance - x:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_6.setText(QtGui.QApplication.translate("genbeam", "Emittacne - y:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_7.setText(QtGui.QApplication.translate("genbeam", "Transverse - x:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_8.setText(QtGui.QApplication.translate("genbeam", "Transverse - y:", None, QtGui.QApplication.UnicodeUTF8))
        self.label_9.setText(QtGui.QApplication.translate("genbeam", "Peak current:", None, QtGui.QApplication.UnicodeUTF8))

class makeund(QtGui.QWidget):
    def __init__(self):
        QtGui.QWidget.__init__(self)
        self.ui = Ui_genund()
        self.ui.setupUi(self)

class makefodo(QtGui.QWidget):
    def __init__(self):
        QtGui.QWidget.__init__(self)
        self.ui = Ui_genfodo()
        self.ui.setupUi(self)

class makebeam(QtGui.QWidget):
    def __init__(self):
        QtGui.QWidget.__init__(self)
        self.ui = Ui_genbeam()
        self.ui.setupUi(self)
